frontend/docsettings.lua

```diff
@@ -246,77 +242,47 @@ function DocSettings:flush()
             ffiutil.fsyncOpenedFile(f_out) -- force flush to the storage device
```

Comment:
Minor (?) issue with koreader/sidecars/: we won't be fsync'ing the intermediate directories created by `util.makePath(s[1])`.
(Which might be a non-issue, as we only create them the first time, and it's not really an issue if the device soon after crash and we lose nearly new docsettings - it only matters for later flush when the user had the time to make some highlights, etc...)

Reply:
Is it safer to replace `util.makePath` with `mkdir -p`?

Reply:
Safer in what way ?
I don't think `mkdir -p` would do any `sync`.
Moreover, there's more risks executing other programs (like launching `mkdir`) than doing things internally from the koreader process in Lua, because executing needs more memory and so may fail if not able to get it. See https://github.com/koreader/koreader/issues/2239#issuecomment-272188927